modules/custom-tuning-specification.adoc

```diff
@@ -43,43 +44,81 @@ profile:
-  <match>                             # an optional array
-  priority: <priority>                # profile ordering priority, lower numbers mean higher priority (0 is the highest priority)
-  profile: <tuned_profile_name>       # e.g. tuned_profile_1
-
+<recommend-item-1>
 # ...
-
-- match:
-  <match>
-  priority: <priority>
-  profile: <tuned_profile_name>       # e.g. tuned_profile_n
+<recommend-item-n>
 ----
 
-If `<match>` is omitted, a profile match (for example, `true`) is assumed.
+The individual items of the list:
+
+[source, yaml]
+----
+- machineConfigLabels: <1>
+    <mcLabels> <2>
+  match: <3>
+  <match> <4>
+  priority: <priority> <5>
+  profile: <tuned_profile_name> <6>
+----
+<1> Optional.
+<2> A dictionary of key/value MachineConfig labels. The keys must be unique.
+<3> If omitted, profile match is assumed unless a profile with a higher priority matches first or `machineConfigLabels` is set.
+<4> An optional list.
+<5> Profile ordering priority. Lower numbers mean higher priority (0 is the highest priority).
+<6> A tuned profile to apply on a match. For example `tuned_profile_1`.
 
-`<match>` is an optional array recursively defined as follows:
+`<match>` is an optional list recursively defined as follows:
 
+[source, yaml]
 ----
-- label: <label_name>     # node or pod label name
-  value: <label_value>    # optional node or pod label value; if omitted, the presence of <label_name> is enough to match
-  type: <label_type>      # optional node or pod type ("node" or "pod"); if omitted, "node" is assumed
-  <match>                 # an optional <match> array
+- label: <label_name> <1>
+  value: <label_value> <2>
+  type: <label_type> <3>
+  <match> <4>
 ----
+<1> Node or Pod label name.
+<2> Optional node or pod label value. If omitted, the presence of `<label_name>` is enough to match.
+<3> Optional node or pod type ("node" or "pod"). If omitted, "node" is assumed.
+<4> An optional `<match>` list.
 
 If `<match>` is not omitted, all nested `<match>` sections must also evaluate to
 `true`. Otherwise, `false` is assumed and the profile with the respective
 `<match>` section will not be applied or recommended. Therefore, the nesting
 (child `<match>` sections) works as logical AND operator. Conversely, if any
-item of the `<match>` array matches, the entire `<match>` array evaluates to
-`true`. Therefore, the array acts as logical OR operator.
-
-.Example
-
+item of the `<match>` list matches, the entire `<match>` list evaluates to
+`true`. Therefore, the list acts as logical OR operator.
+
+If `machineConfigLabels` is defined, MachineConfigPool based matching is turned on
+for the given `recommend:` list item. `<mcLabels>` specifies the labels
+for a MachineConfig. The MachineConfig is created automatically to apply
+additional host settings (for example, kernel boot parameters) profile `<tuned_profile_name>`
```

Comment:
The MachineConfig is created automatically to apply host settings, such as kernel boot parameters, for the profile `<tuned_profile_name>`. You can only apply these settings by creating a MachineConfig.

Reply:
Thanks for the suggestion.  I've taken the first sentence.  The seconds is slightly misleading.  Users will not create MachineConfigs directly, NTO creates those for the users.  What I was trying to convey is that NTO is creating MachineConfigs for host settings that cannot/should_not be performed in any other way. This is useful mostly for scenarios requiring host reboots (e.g. setting boot-time kernel parameters).  I believe that inclusion of the second sentence would not help, rather the opposite.  My original sentence was too long and cumbersome.  Let's keep this brief.  If you have other suggestions for improvements, happy to add them.  This paragraph is key for users to understand!

Reply:
@jmencak This reads much clearer. Thanks!